### PR TITLE
Add autofix to *-no-vendor-prefix

### PR DESCRIPTION
**Pull request by gucong3000**
> Which issue, if any, is this issue related to?

[#2467](https://github.com/stylelint/stylelint/pull/2467)

> Is there anything in the PR that needs further explanation?

Autofixing for rules:
- value-no-vendor-prefix
- property-no-vendor-prefix
- selector-no-vendor-prefix
- media-feature-name-no-vendor-prefix
- at-rule-no-vendor-prefix

Update message for `media-feature-name-no-vendor-prefix`

### CODE REVIEW

**Comment by hudochenkov** [comment]
Thank you! It's a huge work.

Wish you've created an issue with discussion before making PR. I think autofix these rules is too dangerous. Because it will delete user's code. If user need those prefixes, but haven't configured stylelint properly to ignore piece of code with them. It will delete the code, and user might not be aware of these changes.

I think it's better to make things safer and apply autofixing for stylistic rules only for now. But it's possible to extract these autofixes to plugin.

**Comment by gucong3000** [comment]
@hudochenkov 

I think I can adjust my policy and not delete user's code any more

**Comment by hudochenkov** [comment]
@gucong3000 what it will fix then?

**Comment by gucong3000** [comment]
@hudochenkov just unprefix

Input:
```css
a { -webkit-transform: scale(1); transform: scale(1); }```

output:
```css
a { transform: scale(1); transform: scale(1); }```

**Comment by hudochenkov** [comment]
It's not a good solution — make a duplicated code. It's creating more warnings. `declaration-block-no-duplicate-properties` and `no-duplicate-selectors` don't like duplication.

**Comment by gucong3000** [comment]
Should I add a link to [postcss-unprefix](https://github.com/gucong3000/postcss-unprefix) on each documents of these rules?

**Comment by gucong3000** [comment]
@hudochenkov Should we define the level of autofixes?

**Comment by hudochenkov** [comment]
@gucong3000 we are in a [process to define](https://github.com/stylelint/stylelint/issues/2532) what could be fixed.

**Comment by ntwb** [comment]
@gucong3000 Can you rebase this please?

**Comment by gucong3000** [comment]
> Can you rebase this please?

Done.

**Comment by jeddy3** [comment]
@gucong3000 Thanks! 

Are there still reservations around this PR, or can it be merged?

**Comment by gucong3000** [comment]
> add `fix: true` to every `testRule` in the following files, please:

Done.

> I noticed this type of condition in every rule after unprefixing happend:

For future new properties and values that `postcss-unprefix` does not support.

> Are there tests for which covers these conditions?
Not full coverage.

File: lib/rules/property-no-vendor-prefix/__tests__/index.js [hunk 1]
```diff
@@ -107,39 +117,48 @@ testRule(rule, {
     true,
     { ignoreProperties: ["transform", "columns", "/^animation-/i"] }
   ],
+  fix: true,
 
   accept: [
     {
-      code: "a { -webkit-transform: scale(1); }"
+      code: "a { -webkit-transform: scale(1); }",
+      fixed: "a { -webkit-transform: scale(1); }",
```
Comment (hudochenkov) [inline lib/rules/property-no-vendor-prefix/__tests__/index.js]: `accept` tests don't need to have `fixed`.

File: lib/rules/value-no-vendor-prefix/__tests__/index.js [hunk 2]
```diff
@@ -131,33 +143,40 @@ testRule(rule, {
 testRule(rule, {
   ruleName,
   config: [true, { ignoreValues: ["grab", "max-content", "/^linear-/"] }],
+  fix: true,
 
   accept: [
     {
-      code: "cursor: -webkit-grab;"
+      code: "cursor: -webkit-grab;",
+      fixed: "cursor: -webkit-grab;",
```
Comment (hudochenkov) [inline lib/rules/value-no-vendor-prefix/__tests__/index.js]: `accept` tests don't need to have `fixed`.

File: lib/rules/media-feature-name-no-vendor-prefix/index.js [hunk 3]
```diff
@@ -24,21 +28,34 @@ const rule = function(actual) {
...
+        }
+      }
+      do {
         report({
-          message: messages.rejected,
+          message: messages.rejected(match[0]),
           node: atRule,
-          word: match,
+          index: match.index,
+          // word: match[0],
```
Comment (hudochenkov) [inline lib/rules/media-feature-name-no-vendor-prefix/index.js]:
```suggestion
```

**Comment by hudochenkov** [comment]
Closing due inactivity. We can reopen if requested changes were made.

**Pull request closed without merging**